File: java/sort_colors/sort_colors.java
import java.util.Arrays;

class Solution {
    public void sortColors(int[] nums) {
        int[] counts = { 0, 0, 0 };

        for (int color : nums) {
            counts[color]++;
        }

        int R = counts[0];
        int W = counts[1];
        int B = counts[2];

        Arrays.fill(nums, 0, R, 0);
        Arrays.fill(nums, R, R + W, 1);
        Arrays.fill(nums, R + W, R + W + B, 2);
    }
}

File: c++/sort_colors/sort_colors.cpp
#include <vector>
#include <iostream>
#include <algorithm>
#include <bits/stdc++.h>
#include <array>

using namespace std;

class Solution
{
public:
    void sortColors(vector<int> &nums)
    {
        int counts[3] = {0, 0, 0};

        for (auto &&color : nums)
        {
            counts[color]++;
        }

        int R = counts[0];
        int W = counts[1];
        int B = counts[2];

        fill(nums.begin(), nums.begin() + R, 0);
        fill(nums.begin() + R, nums.begin() + R + W, 1);
        fill(nums.begin() + R + W, nums.end(), 2);
    }
};